lib/table_of_contents/parser.rb

```diff
@@ -13,13 +13,58 @@ def initialize(html)
       def build_toc
         toc = %Q{<ul class="section-nav">\n}
 
+        min_h_num = 6
         @entries.each do |entry|
-          toc << %Q{<li class="toc-entry toc-#{entry[:node_name]}"><a href="##{entry[:id]}#{entry[:uniq]}">#{entry[:text]}</a></li>\n}
+          h_num = entry[:node_name][1].to_i
+          min_h_num = [min_h_num, h_num].min
         end
+        toc << build_lis(@entries, min_h_num)
 
         toc << '</ul>'
       end
 
+      # Returns the list items for entries
+      def build_lis(entries, min_h_num)
```

Comment:
This logic generates html below:

```html
<ul class="section-nav">
  <li class="toc-entry toc-h2">base 1 </li>
  <ul>
    <li class="toc-entry toc-h3">nested 1</li>
    <li class="toc-entry toc-h3">nested 2</li>
  </ul>
</ul>
```

However, preferred html is like:

```html
<ul class="section-nav">
  <li class="toc-entry toc-h2">base 1 
    <ul>
      <li class="toc-entry toc-h3">nested 1</li>
      <li class="toc-entry toc-h3">nested 2</li>
    </ul>
  </li>
</ul>
```

Could you modify the logic to create such nested toc-list? (or is it diffecult??)

see also. [Proper way to make HTML nested list? - Stack Overflow](http://stackoverflow.com/questions/5899337/proper-way-to-make-html-nested-list)